applications/StructuralMechanicsApplication/custom_conditions/moving_load_condition.cpp

```diff
@@ -0,0 +1,270 @@
+template< std::size_t TDim, std::size_t TNumNodes >
+Condition::Pointer MovingLoadCondition< TDim, TNumNodes>::Clone (
+    IndexType NewId,
+    NodesArrayType const& ThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    Condition::Pointer p_new_cond = Kratos::make_intrusive<MovingLoadCondition>(NewId, GetGeometry().Create(ThisNodes), pGetProperties());
+    p_new_cond->SetData(this->GetData());
+    p_new_cond->Set(Flags(*this));
+    return p_new_cond;
+
+    KRATOS_CATCH("");
+}
+
+//******************************* DESTRUCTOR *****************************************
+//************************************************************************************
+template< std::size_t TDim, std::size_t TNumNodes >
+MovingLoadCondition< TDim, TNumNodes>::~MovingLoadCondition()
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+template< std::size_t TDim, std::size_t TNumNodes >
+void MovingLoadCondition< TDim, TNumNodes>::CalculateAll(
+    MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo,
+    const bool CalculateStiffnessMatrixFlag,
+    const bool CalculateResidualVectorFlag
+    )
+{
+    KRATOS_TRY
+
+    const unsigned int NumberOfNodes = GetGeometry().size();
+    const unsigned int Dimension = GetGeometry().WorkingSpaceDimension();
+
+    const unsigned int block_size = this->GetBlockSize();
+    // Resizing as needed the LHS
+    unsigned int MatSize = NumberOfNodes * block_size;
+   /* for (const auto& r_node : GetGeometry().Points()) {
+        if (r_node.HasDofFor(ROTATION_X)) { MatSize += 1; }
+        if (r_node.HasDofFor(ROTATION_Y)) { MatSize += 1; }
+        if (r_node.HasDofFor(ROTATION_Z)) { MatSize += 1; }
+    }*/
+
+
+    if ( CalculateStiffnessMatrixFlag == true ) //calculation of the matrix is required
+    {
+        if ( rLeftHandSideMatrix.size1() != MatSize )
+        {
+            rLeftHandSideMatrix.resize( MatSize, MatSize, false );
+        }
+
+        noalias( rLeftHandSideMatrix ) = ZeroMatrix( MatSize, MatSize ); //resetting LHS
+    }
+
+    //resizing as needed the RHS
+    if ( CalculateResidualVectorFlag == true ) //calculation of the matrix is required
+    {
+        if ( rRightHandSideVector.size( ) != MatSize )
+        {
+            rRightHandSideVector.resize( MatSize, false );
+        }
+
+        noalias( rRightHandSideVector ) = ZeroVector( MatSize ); //resetting RHS
+    }
+
+    // Vector with a loading applied to the condition
+    array_1d<double, 3 > MovingLoad = ZeroVector(3);
+    if( this->Has(MOVING_LOAD ) )
+    {
+        noalias(MovingLoad) = this->GetValue( MOVING_LOAD );
+    }
+
+    // check if cond should be calculated
+    bool is_moving_load = false;
+    for (int i = 0; i < TDim; ++i)
+    {
+        if (abs(MovingLoad[i]) > DBL_EPSILON)
```

Comment:
```suggestion
        if (std::abs(MovingLoad[i]) > DBL_EPSILON)
```
This is very dangerous. abs returns a integer, 0.01 will be 0